api/logic/modplatform/atlauncher/ATLPackInstallTask.cpp

```diff
@@ -0,0 +1,416 @@
+#include <QtXml/QDomDocument>
+#include <Env.h>
+#include <quazip.h>
+#include <QtConcurrent/QtConcurrent>
+#include <MMCZip.h>
+#include <minecraft/OneSixVersionFormat.h>
+#include "ATLPackInstallTask.h"
+
+#include "BuildConfig.h"
+#include "FileSystem.h"
+#include "Json.h"
+#include "minecraft/MinecraftInstance.h"
+#include "minecraft/PackProfile.h"
+#include "settings/INISettingsObject.h"
+#include "meta/Index.h"
+#include "meta/VersionList.h"
+
+namespace ATLauncher {
+
+PackInstallTask::PackInstallTask(QString pack, QString version)
+{
+    m_pack = pack;
+    m_version_name = version;
+}
+
+bool PackInstallTask::abort()
+{
+    return true;
+}
+
+void PackInstallTask::executeTask()
+{
+    auto *netJob = new NetJob("ATLauncher::VersionFetch");
+    auto searchUrl = QString(BuildConfig.ATL_DOWNLOAD_SERVER + "packs/%1/versions/%2/Configs.xml")
```

Comment:
Any reason you're using the XML and not the json? We do not use the XML and do a lot of transformation server side to transform this xml to json.

I'll be removing this xml file from the download server fairly soon

Reply:
Example: https://download.nodecdn.net/containers/atl/packs/SkyFactory4/versions/4.2.2/Configs.json

Reply:
It doesn't make a lot of difference either way, if the XML is going - I'll move over to the JSON.

My concern was that the JSON seemed to be tied to launcher implementation details, and some data from the XML could be omitted as a result. This was something I meant to discuss with you, just something I forgot about.

Reply:
Yeah I've been meaning to remove the XML for years, and there's a big overhaul of the backend systems coming within the next month, which includes removing these files.

All the same information should be there. There shouldn't be anything missing

Reply:
Resolved by https://github.com/MultiMC/MultiMC5/pull/3314/commits/44a12a9836eb692f2ad23ca03a016bb67b6d616f.